parsers/BG.py

```diff
@@ -1,72 +1,109 @@
 from logging import Logger, getLogger
 
 from requests import Session
 
-TYPE_MAPPING = {  # Real values around midnight
-    "АЕЦ": "nuclear",  # 2000
-    "Кондензационни ТЕЦ": "coal",  # 1800
-    "Топлофикационни ТЕЦ": "gas",  # 146
-    "Заводски ТЕЦ": "gas",  # 147
-    "ВЕЦ": "hydro",  # 7
-    "Малки ВЕЦ": "hydro",  # 74
-    "ВяЕЦ": "wind",  # 488
-    "ФЕЦ": "solar",  # 0
-    "Био ТЕЦ": "biomass",  # 29
-    "Био ЕЦ": "biomass",  # 29
-    "Товар РБ": "consumption",  # 3175
+from electricitymap.contrib.lib.models.event_lists import ProductionBreakdownList
+from electricitymap.contrib.lib.models.events import ProductionBreakdown, ProductionMix
+from electricitymap.contrib.lib.types import ZoneKey
+from electricitymap.contrib.parsers.lib.exceptions import ParserException
+
+PARSER = "BG.py"
+
+SOURCE = "eso.bg"
+SOURCE_API_URL = "http://www.eso.bg/api/rabota_na_EEC_json.php"
+
+PRODUCTION_TYPES_BG_TO_EN = {
+    "АЕЦ": "nuclear",
+    "Кондензационни ТЕЦ": "coal",
+    "Топлофикационни ТЕЦ": "gas",
+    "Заводски ТЕЦ": "gas",
+    "ВЕЦ": "hydro",
+    "Малки ВЕЦ": "hydro",
+    "ВяЕЦ": "wind",
+    "ФЕЦ": "solar",
+    "Био ТЕЦ": "biomass",
+    "Био ЕЦ": "biomass",
+    "Товар РБ": "consumption",
 }
 
 
-def fetch_production(
-    zone_key: str = "BG",
-    session: Session | None = None,
-    target_datetime=None,
-    logger: Logger = getLogger(__name__),
-) -> dict:
-    """Requests the last known production mix (in MW) of a given country."""
-    if target_datetime:
-        raise NotImplementedError("This parser is not yet able to parse past dates")
-
-    r = session or Session()
-    url = "http://www.eso.bg/api/rabota_na_EEC_json.php"
-    res = r.get(url)
-
-    assert (
-        res.status_code == 200
-    ), f"Exception when fetching production for {zone_key}: error when calling url={url}"
-
-    response = res.json()
-
-    logger.debug(f"Raw generation breakdown: {response}")
-
-    datapoints = []
-    for row in response:
-        for k in TYPE_MAPPING:
-            if row[0].startswith(k):
-                datapoints.append((TYPE_MAPPING[k], row[1]))
-                break
+def _fetch_all_production_events(
+    zone_key: ZoneKey,
+    session: Session,
+    target_datetime: datetime | None,
+    logger: Logger,
+) -> list[ProductionBreakdown]:
+    """Fetches all production events from the backend source API."""
+    if target_datetime is not None:
+        raise ParserException(
+            PARSER, "This parser is not yet able to parse historical data", zone_key
+        )
+
+    time = datetime.now(timezone.utc)
+    response = session.get(SOURCE_API_URL)
+    if not response.status_code == 200:
+        raise ParserException(
+            PARSER,
+            f"Exception when fetching production error code: {response.status_code}: {response.text}",
+            zone_key,
+        )
+
+    response_payload = response.json()
+    logger.debug(f"Raw generation breakdown: {response_payload}")
 
     production = {}
```

Comment:
This should be a production mix object instead:
```suggestion
    production = ProductionMix()
```
But we also need a ProductionBreakdownList object here as well so we can append it to something.